scala3doc/src/main/scala/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -0,0 +1,205 @@
+import org.jetbrains.dokka.model.properties.{WithExtraProperties}
+import java.util.{List => JList}
+
+import scala.tasty.Reflection
+import scala.tasty.inspector.TastyInspector
+import dotty.dokka.model.api.withNewMembers
+import com.virtuslab.dokka.site.SourceSetWrapper
+
+/** Responsible for collectively inspecting all the Tasty files we're interested in.
+  *
+  * Delegates most of the work to [[TastyParser]] [[dotty.dokka.tasty.TastyParser]].
+  */
+case class DokkaTastyInspector(sourceSet: SourceSetWrapper, parser: Parser, config: DottyDokkaConfig) extends DokkaBaseTastyInspector with TastyInspector
+
+import dotty.tools.dotc.core.Contexts.{Context => DottyContext}
+case class SbtDokkaTastyInspector(
+  sourceSet: SourceSetWrapper,
+  config: DottyDokkaConfig,
+  filesToDocument: List[String],
+  rootCtx: DottyContext,
+) extends DokkaBaseTastyInspector:
+  self =>
+
+  import dotty.tools.dotc.Compiler
+  import dotty.tools.dotc.Driver
+  import dotty.tools.dotc.Run
+  import dotty.tools.dotc.core.Contexts.Context
+  import dotty.tools.dotc.core.Mode
+  import dotty.tools.dotc.core.Phases.Phase
+  import dotty.tools.dotc.fromtasty._
+  import dotty.tools.dotc.quoted.QuoteContextImpl
+
+  val parser: Parser = null
+
+  def run(): List[DPackage] = {
+    val driver = new InspectorDriver
+    driver.run(filesToDocument)(rootCtx)
+    result()
+  }
+
+  class InspectorDriver extends Driver:
+    override protected def newCompiler(implicit ctx: Context): Compiler = new TastyFromClass
+
+    def run(filesToDocument: List[String])(implicit ctx: Context): Unit =
+      doCompile(newCompiler, filesToDocument)
+
+  end InspectorDriver
+
+  class TastyFromClass extends TASTYCompiler:
+
+    override protected def frontendPhases: List[List[Phase]] =
+      List(new ReadTasty) :: // Load classes from tasty
+      Nil
+
+    override protected def picklerPhases: List[List[Phase]] = Nil
+
+    override protected def transformPhases: List[List[Phase]] = Nil
+
+    override protected def backendPhases: List[List[Phase]] =
+      List(new TastyInspectorPhase) ::  // Print all loaded classes
+      Nil
+
+    override def newRun(implicit ctx: Context): Run =
+      reset()
+      new TASTYRun(this, ctx.fresh.addMode(Mode.ReadPositions).addMode(Mode.ReadComments))
+
+  end TastyFromClass
+
+  class TastyInspectorPhase extends Phase:
+
+    override def phaseName: String = "tastyInspector"
+
+    override def run(implicit ctx: Context): Unit =
+      val qctx = QuoteContextImpl()
+      self.processCompilationUnit(using qctx)(ctx.compilationUnit.tpdTree.asInstanceOf[qctx.tasty.Tree])
+
+  end TastyInspectorPhase
+
+end SbtDokkaTastyInspector
+
+trait DokkaBaseTastyInspector:
+  val sourceSet: SourceSetWrapper
+  val parser: Parser
+  val config: DottyDokkaConfig
+
+  private val topLevels = Seq.newBuilder[Documentable]
+
+  def processCompilationUnit(using ctx: quoted.QuoteContext)(root: ctx.reflect.Tree): Unit =
```

Comment:
```suggestion
  def processCompilationUnit(using QuoteContext)(root: qctx.reflect.Tree): Unit =
```

`qctx` acts like the `ctx` in the compiler. You have to import `scala.quote._` first.

Reply:
Adressed in 78b7d49 in #10092